### DISCUSSION TITLE
NeighborLoader imbalanced sampling question

### DISCUSSION THREAD

**Discussion by EvanUp** [pyg-team/pytorch_geometric#3784] [Q&A] [answered]
Hey, I have two loader questions:

1) If num_neighbors is [30] and some nodes in a batch have fewer than 30 neighbors, how does NeighborLoader maintain consistent batch sizes?

2) Do any loaders in PyG support oversampling nodes in cases where node classes are imbalanced?

**Comment by rusty1s** [comment, marked as answer]
Hi and sorry for the late reply.

1. `NeighborLoader` does not oversample neighbors. If there  exists fewer than 30 neighbors for a particular node, its full neighborhood will be sampled.
2. Similar to vanilla PyTorch, you should be able to pass in a custom sampler for oversampling/undersampling, see [here](https://discuss.pytorch.org/t/balanced-sampling-between-classes-with-torchvision-dataloader/2703/30?page=2). Let me know if that works for you.

**Reply by EvanUp** [reply]
Thank you! Really appreciate the response and the library! I hadn't realized NeighborLoader could able node indices and not just booleans - that made everything a lot easier! I should read the documentation more thoroughly next time :)
